crates/pathfinder/src/bin/pathfinder/config.rs

```diff
@@ -618,6 +627,47 @@ enum RpcCorsDomainsParseError {
     WildcardAmongOtherValues,
 }
 
+fn parse_versioned_constants(
+    path: PathBuf,
+) -> Result<VersionedConstants, ParseVersionedConstantsError> {
+    let file = File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let versioned_constants = serde_json::from_reader(reader)?;
+
+    Ok(versioned_constants)
+}
+
+pub fn parse_versioned_constants_or_exit(path: PathBuf) -> VersionedConstants {
+    use clap::error::ErrorKind;
+
+    match parse_versioned_constants(path) {
+        Ok(versioned_constants) => versioned_constants,
+        Err(error) => Cli::command()
+            .error(ErrorKind::ValueValidation, error)
+            .exit(),
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum ParseVersionedConstantsError {
+    #[error("IO error while reading versioned constants: {0}.")]
+    Io(std::io::Error),
```

Comment:
You can use the `#[from]` directive in both cases, [ref](https://docs.rs/thiserror/latest/thiserror/#example).

Reply:
Thanks, fixed.